reporter/internal/pdata/generate.go

```diff
@@ -216,17 +221,8 @@ func (p *Pdata) setProfile(
 					}
 					locInfo.functionIndex = funcSet.Add(fi)
 				}
-
-				idx, exists := mappingSet.AddWithCheck(traceInfo.Files[i])
-				locInfo.mappingIndex = idx
-				if !exists {
-					// To be compliant with the protocol, generate a dummy mapping entry.
-					mapping := dic.MappingTable().AppendEmpty()
-					mapping.SetFilenameStrindex(stringSet.Add(""))
-					attrMgr.AppendOptionalString(mapping.AttributeIndices(),
-						semconv.ProcessExecutableBuildIDHtlhashKey,
-						traceInfo.Files[i].StringNoQuotes())
-				}
+				// mapping_table[0] is always the dummy mapping
+				locInfo.mappingIndex = 0
```

Comment:
Further up, for native frames, when adding a new Mapping, `locInfo.mappingIndex` isn't set.

Reply:
Hm, for me - when viewing the code in Github - it is unexpected that AddWithCheck() returns a valid locationMappingIndex. Actually it does, but for that, you have to open helper.go and take a look at AddWithCheck(). It's a deviating pattern from e.g. `v, ok := m["foo"]` and makes things harder to follow.
Anyway, forget about my initial comment.